### ISSUE TITLE
nested readme file uploads not working 

### ISSUE THREAD

**Issue by bir87** [justmiles/go-markdown2confluence#25]
markdown2confluence --space 'MyTeamSpace' --parent 'API Docs' markdown-files

I am guessing this should upload all the nested md files present in the dir "markdown-files"? Am i correct.

**Comment by bir87** [comment]
Error updating content: Version must be incremented on update. Current version is: 95

**Comment by justmiles** [comment]
You are correct. Nested markdown files will be upload under the parent page "API Docs" in the "MyTeamSpace" space.

Unfortunately, nested files with the same name aren't considered different files to Confluence. We're determining the remote page name by matching only the space and filename. So two files named `readme.md` in different directories are updating the same page in Confluence.

Solving for #14 also requires persisting the file and remote ID locally and having that mapping would subsequently fix this problem too.

Up to this point I've just had unique names for my docs. I realize that's not a real solution, though. I'll probably take a stab at #14 before this one, but happy to accept PRs.

**Comment by bir87** [comment]
Does markdown2confluence upload all these files under 1 parent so it will create the same tree strucutre under that parent in confluence, so lets say if i have a parent named "test" in confluence will it consist the same tree str as below.

test
├── readme02.md
├── test-01
│   ├── readme1.md
│   └── test-02
│       └── readme-03.md
└── test.md

**Comment by justmiles** [comment]
Yes, it mirrors the folder structure.

**Comment by justmiles** [comment]
For directories it will create a single page with links to the child pages. It might be a good idea to earmark readme.md or index.md files as the content for those base directory pages.

**Comment by elijahgagne** [comment]
I like using README.md for this purpose because that's what a lot of Git providers (e.g. GitHub, GitLab, Bitbucket) will display for the content of a folder. A while back I opened up https://github.com/justmiles/go-markdown2confluence/issues/17. Back in May, in a fork, I implemented the feature to work for my purposes at https://github.com/justmiles/go-markdown2confluence/commit/beb13e3ea769c0fab54a563875d31cdb6d465375. 

I question if this is good enough for a pull request or perhaps a starting point for someone else to improve. If you have any suggestions, I'd be happy to work on it a little and submit a PR.

**Comment by justmiles** [comment]
Hey that's great. If it's worked for you I'm sure it'll work for others. I need to write some unit tests so that people feel more comfortable about submitting PRs but this project hasn't been a high priority. In any case submit a PR! I'm sure others will benefit from it.

**Comment by bir87** [comment]
i am not able to replciate this tree structure  below in confluence.
test
├── readme02.md
├── **test-01**
│   ├── readme1.md
│   └── **test-02**
│   └── readme-03.md
└── test.md

My output is : everything gets put in the parent dir, the sub-dirs dont get created.

**test**
├readme02.md
│readme1.md
│readme-03.md
test.md

**Comment by elijahgagne** [comment]
It seems to work for me. 

```
# Create structure
mkdir test
mkdir test/test-01
echo 'readme02.md' > test/readme02.md
echo 'test.md' > test/test.md
echo 'readme1.md' > test/test-01/readme1.md
echo 'test-02' > test/test-01/test-02
echo 'readme-03.md' > test/test-01/readme-03.md
```

```
# Run sync
markdown2confluence -s '~293896096' /tmp/test
```

You'll note that `test-02` doesn't come over because it doesn't have a `.md` extension.

![image](https://user-images.githubusercontent.com/2213005/93246815-564a0e80-f75b-11ea-86a7-fce738dea761.png)

**Comment by bir87** [comment]
is there a limit on how big the directory structure is ? Because it does not work for me like yours.

**Comment by elijahgagne** [comment]
None that I know of. Just to be clear, you're using the latest version, follow the commands I pasted above, and still get a different result?

**Comment by bir87** [comment]
so if this page called abs.md already exists in space XYZ .. i cant create a different parent and expect abs.md to be under that parent correct. .. since it already sees inside space XYZ it will just go ahead and update it rather than recreate

**Comment by elijahgagne** [comment]
Yes, that's my understanding. This stems from the fact the Confluence does not support two pages with the same name.

**Comment by bir87** [comment]
it looks like it does work for small nested tree structures with few kb of .md files. But will not work for a vast tree structure with multiple md files in each layer. 
what is the rest api you are using to create spaces or folder in confluence.

**Comment by elijahgagne** [comment]
I am working in a Confluence environment with about 20 people and only a few of us are syncing our changes. I generally write a few pages and then sync them. So I never have a lot of new content that needs to get synced.
